rubix-core/src/test/java/com/qubole/rubix/core/TestUpdatingofCache.java

```diff
@@ -0,0 +1,206 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. See accompanying LICENSE file.
+ */
+package com.qubole.rubix.core;
+
+import com.qubole.rubix.bookkeeper.BookKeeperConfig;
+import com.qubole.rubix.bookkeeper.BookKeeperServer;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.SequenceFile;
+import org.apache.hadoop.io.UTF8;
+import org.apache.hadoop.io.file.tfile.ByteArray;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.RandomAccessFile;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+import static org.testng.AssertJUnit.assertTrue;
+
+/**
+ * Created by qubole on 19/7/16 .
+ */
+public class TestUpdatingofCache
+{
+    int blockSize = 100;
+    String backendFileName = "/tmp/backendFile";
+    Path backendPath = new Path("file://" + backendFileName);
+
+    CachingInputStream inputStream;
+
+    private static final Log log = LogFactory.getLog(TestUpdatingofCache.class);
+
+    @BeforeMethod
+    public void setup_initial()
+            throws IOException, InterruptedException
+    {
+        final Configuration conf = new Configuration();
+        conf.setBoolean(CachingConfigHelper.DATA_CACHE_STRICT_MODE, true);
+        conf.setInt(BookKeeperConfig.DATA_CACHE_BOOKKEEPER_PORT, 3456);
+        Thread thread = new Thread() {
+            public void run()
+            {
+                BookKeeperServer.startServer(conf);
+            }
+        };
+        thread.start();
+
+        DataGen.populateFile(backendFileName);
+        LocalFSInputStream localFSInputStream = new LocalFSInputStream(backendFileName);
+        FSDataInputStream fsDataInputStream = new FSDataInputStream(localFSInputStream);
+
+        File file = new File(backendFileName);
+        conf.setInt(BookKeeperConfig.BLOCK_SIZE, blockSize);
+
+        while (!BookKeeperServer.isServerUp()) {
+            Thread.sleep(200);
+            log.info("Waiting for BookKeeper Server to come up");
+        }
+        log.info("All set to test");
+
+        // This should be after server comes up else client could not be created
+        inputStream = new CachingInputStream(fsDataInputStream, conf, backendPath, file.length(),file.lastModified(), new CachingFileSystemStats());
+    }
+
+    public void setup()
+            throws InterruptedException, IOException
+    {
+        final Configuration conf = new Configuration();
```

Comment:
The initial setup starts the bookkeeper service and populates the backend file. We do not want these steps to repeat when we start a new CachingInputStream.